src/Orchard/Mvc/ViewEngines/Razor/WebViewPage.cs

```diff
@@ -190,11 +190,10 @@ public abstract class WebViewPage<TModel> : System.Web.Mvc.WebViewPage<TModel>,
                 _tenantPrefix = WorkContext.Resolve<ShellSettings>().RequestUrlPrefix ?? "";
             }
 
-            if (!String.IsNullOrEmpty(_tenantPrefix)
-                && path.StartsWith("~/")  
-                && !CommonLocations.Any(gpp=>path.StartsWith(gpp, StringComparison.OrdinalIgnoreCase))
-            ) { 
-                    return base.Href("~/" + _tenantPrefix + path.Substring(2), pathParts);
+            if (!string.IsNullOrWhiteSpace(_tenantPrefix)
+                && path.StartsWith("~/")
+                && !CommonLocations.Any(gpp => path.StartsWith(gpp, StringComparison.OrdinalIgnoreCase))) {
+                return base.Href("~/" + _tenantPrefix + path.Substring(1), pathParts);
```

Comment:
Why is the substring start index parameter changed?
Since code gets to line 196 only if path starts with "~/", it stripped "~/" in the previous version of the file; now it only strips the "~" character: is it correct keeping the "/" character?

Reply:
FYI this is only applied if there is a tenant prefix. e.g., the base URL is `mysite.com/mytenant`. In this case you still need the `/` between the tenant prefix and the rest of the path.